benchmarks/src/fill.cpp

```diff
@@ -0,0 +1,127 @@
+#include <benchmark/benchmark.h>
+#include <cstring>
+#include <memory>
+#include <ranges>
+#include <vector>
+namespace {
+    template <typename Contained, Contained Value>
+    void handwritten_loop(benchmark::State& state) {
+        const size_t r0 = static_cast<size_t>(state.range(0));
+        std::unique_ptr<Contained[]> buffer(new Contained[r0]);

```

Comment:
Any advantage of using `new` and not `make_unique` ?

Reply:
You might want to use `vector` with `highly_aligned_allocator` / `not_highly_aligned_allocator` to make allocation alignment more predictable. You can have `.data()` for pointer. Currently it can be 16 bytes (`malloc` alignment), or 32 (accidentally) or more (there's a chance)

Reply:
No particular reason. I think I'd need `make_unique_for_overwrite` I can switch over.

I used unique_ptr just to avoid double initializing things, even though that's not really a big deal. I didn't know about those allocators.

I guess it's probably best to switch between highly/not-highly aligned for different variants, instead of just 2x-ing the number of tests.